crawl.go

```diff
@@ -0,0 +1,198 @@
+	"/dns4/bootstrap-2.mainnet.filops.net/tcp/1347/p2p/12D3KooWEWVwHGn2yR36gKLozmb4YjDJGerotAPGxmdWZx2nxMC4",
+	"/dns4/bootstrap-3.mainnet.filops.net/tcp/1347/p2p/12D3KooWKhgq8c7NQ9iGjbyK7v7phXvG6492HQfiDaGHLHLQjk7R",
+	"/dns4/bootstrap-4.mainnet.filops.net/tcp/1347/p2p/12D3KooWL6PsFNPhYftrJzGgF5U18hFoaVhfGk7xwzD8yVrHJ3Uc",
+	"/dns4/bootstrap-5.mainnet.filops.net/tcp/1347/p2p/12D3KooWLFynvDQiUpXoHroV1YxKHhPJgysQGH2k3ZGwtWzR4dFH",
+	"/dns4/bootstrap-6.mainnet.filops.net/tcp/1347/p2p/12D3KooWP5MwCiqdMETF9ub1P3MbCvQCcfconnYHbWg6sUJcDRQQ",
+	"/dns4/bootstrap-7.mainnet.filops.net/tcp/1347/p2p/12D3KooWRs3aY1p3juFjPy8gPN95PEQChm2QKGUCAdcDCC4EBMKf",
+	"/dns4/bootstrap-8.mainnet.filops.net/tcp/1347/p2p/12D3KooWScFR7385LTyR4zU1bYdzSiiAb5rnNABfVahPvVSzyTkR",
+	"/dns4/lotus-bootstrap.ipfsforce.com/tcp/41778/p2p/12D3KooWGhufNmZHF3sv48aQeS13ng5XVJZ9E6qy2Ms4VzqeUsHk",
+	"/dns4/bootstrap-0.starpool.in/tcp/12757/p2p/12D3KooWGHpBMeZbestVEWkfdnC9u7p6uFHXL1n7m1ZBqsEmiUzz",
+	"/dns4/bootstrap-1.starpool.in/tcp/12757/p2p/12D3KooWQZrGH1PxSNZPum99M1zNvjNFM33d1AAu5DcvdHptuU7u",
+	"/dns4/node.glif.io/tcp/1235/p2p/12D3KooWBF8cpp65hp2u9LK5mh19x67ftAam84z9LsfaquTDSBpt",
+	"/dns4/bootstrap-0.ipfsmain.cn/tcp/34721/p2p/12D3KooWQnwEGNqcM2nAcPtRR9rAX8Hrg4k9kJLCHoTR5chJfz6d",
+	"/dns4/bootstrap-1.ipfsmain.cn/tcp/34723/p2p/12D3KooWMKxMkD5DMpSWsW7dBddKxKT7L2GgbNuckz9otxvkvByP",
+}
+
+var CalibnetPeers = []string{
+	"/dns4/bootstrap-0.calibration.fildev.network/tcp/1347/p2p/12D3KooWJkikQQkxS58spo76BYzFt4fotaT5NpV2zngvrqm4u5ow",
+	"/dns4/bootstrap-1.calibration.fildev.network/tcp/1347/p2p/12D3KooWLce5FDHR4EX4CrYavphA5xS3uDsX6aoowXh5tzDUxJav",
+	"/dns4/bootstrap-2.calibration.fildev.network/tcp/1347/p2p/12D3KooWA9hFfQG9GjP6bHeuQQbMD3FDtZLdW1NayxKXUT26PQZu",
+	"/dns4/bootstrap-3.calibration.fildev.network/tcp/1347/p2p/12D3KooWMHDi3LVTFG8Szqogt7RkNXvonbQYqSazxBx41A5aeuVz",
+}
+
+type FilecoinPeer struct {
+	ctx            context.Context
+	dht            *dht.IpfsDHT
+	host           host.Host
+	bootstrapPeers []peer.AddrInfo
+}
+
+func FilecoinBootstrapPeers(peerList []string) []peer.AddrInfo {
+	if peerList == nil {
+		peerList = MainnetPeers
+	}
+
+	maddrs := make([]multiaddr.Multiaddr, len(peerList))
+
+	for i, bp := range peerList {
+		var err error
+		maddrs[i], err = multiaddr.NewMultiaddr(bp)
+		if err != nil {
+			return nil
+		}
+	}
+
+	peers, _ := peer.AddrInfosFromP2pAddrs(maddrs...)
+
+	return peers
+}
+
+func SetupFilecoinPeer(
+	ctx context.Context,
+	network NetworkName,
+) (*FilecoinPeer, error) {
+	if network == "" {
+		network = Mainnet
+	}
+
+	var pp protocol.ID
+	var bootstrapPeers []peer.AddrInfo
+
+	switch network {
+	case Mainnet:
+		pp = "/fil/kad/testnetnet"
+		bootstrapPeers = FilecoinBootstrapPeers(MainnetPeers)
+	case Calibnet:
+		pp = "/fil/kad/calibnet"
+		bootstrapPeers = FilecoinBootstrapPeers(CalibnetPeers)
+	}
+
+	var ddht *dht.IpfsDHT
+	priv, _, err := crypto.GenerateKeyPair(crypto.RSA, 2048)
```

Comment:
It's not important I think, but we tend to use Ed25519 these days.

```suggestion
	priv, _, err := crypto.GenerateKeyPair(crypto.Ed25519, -1)
```

Reply:
(produces smaller keys, which I hope means it is lighter in general).